platform/darwin/src/MGLSymbolStyleLayer.h

```diff
@@ -0,0 +1,307 @@
+    MGLSymbolStyleLayerTextRotationAlignmentMap,
+    MGLSymbolStyleLayerTextRotationAlignmentViewport,
+};
+typedef NS_ENUM(NSUInteger, MGLSymbolStyleLayerTextJustify) {
+    MGLSymbolStyleLayerTextJustifyLeft,
+    MGLSymbolStyleLayerTextJustifyCenter,
+    MGLSymbolStyleLayerTextJustifyRight,
+};
+typedef NS_ENUM(NSUInteger, MGLSymbolStyleLayerTextAnchor) {
+    MGLSymbolStyleLayerTextAnchorCenter,
+    MGLSymbolStyleLayerTextAnchorLeft,
+    MGLSymbolStyleLayerTextAnchorRight,
+    MGLSymbolStyleLayerTextAnchorTop,
+    MGLSymbolStyleLayerTextAnchorBottom,
+    MGLSymbolStyleLayerTextAnchorTopLeft,
+    MGLSymbolStyleLayerTextAnchorTopRight,
+    MGLSymbolStyleLayerTextAnchorBottomLeft,
+    MGLSymbolStyleLayerTextAnchorBottomRight,
+};
+typedef NS_ENUM(NSUInteger, MGLSymbolStyleLayerTextTransform) {
+    MGLSymbolStyleLayerTextTransformNone,
+    MGLSymbolStyleLayerTextTransformUppercase,
+    MGLSymbolStyleLayerTextTransformLowercase,
+};
+typedef NS_ENUM(NSUInteger, MGLSymbolStyleLayerIconTranslateAnchor) {
+    MGLSymbolStyleLayerIconTranslateAnchorMap,
+    MGLSymbolStyleLayerIconTranslateAnchorViewport,
+};
+typedef NS_ENUM(NSUInteger, MGLSymbolStyleLayerTextTranslateAnchor) {
+    MGLSymbolStyleLayerTextTranslateAnchorMap,
+    MGLSymbolStyleLayerTextTranslateAnchorViewport,
+};
+
+@interface MGLSymbolStyleLayer : MGLStyleLayer
+
+#pragma mark - Accessing the Layout Attributes
+
+/**
+  Label placement relative to its geometry. `line` can only be used on LineStrings and Polygons.
+*/
+@property (nonatomic) MGLSymbolStyleLayerSymbolPlacement symbolPlacement;
+
+/**
+  Distance between two symbol anchors.
+*/
+@property (nonatomic) CGFloat symbolSpacing;
+
+/**
+  If true, the symbols will not cross tile edges to avoid mutual collisions. Recommended in layers that don't have enough padding in the vector tile to prevent collisions, or if it is a point symbol layer placed after a line symbol layer.
+*/
+@property (nonatomic) BOOL symbolAvoidEdges;
+
+/**
+  If true, the icon will be visible even if it collides with other previously drawn symbols.
+*/
+@property (nonatomic) BOOL iconAllowOverlap;
+
+/**
+  If true, other symbols can be visible even if they collide with the icon.
+*/
+@property (nonatomic) BOOL iconIgnorePlacement;
+
+/**
+  If true, text will display without their corresponding icons when the icon collides with other symbols and the text does not.
+*/
+@property (nonatomic) BOOL iconOptional;
+
+/**
+  Orientation of icon when map is rotated.
+*/
+@property (nonatomic) MGLSymbolStyleLayerIconRotationAlignment iconRotationAlignment;
+
+/**
+  Scale factor for icon. 1 is original size, 3 triples the size.
+*/
+@property (nonatomic) CGFloat iconSize;
+
+/**
+  Position and scale an icon by the its corresponding text.
+*/
+@property (nonatomic) MGLSymbolStyleLayerIconTextFit iconTextFit;
+
+/**
+  Size of padding area around the text-fit size in clockwise order: top, right, bottom, left.
```

Comment:
If we do adopt `NSEdgeInsets`/`UIEdgeInsets` for properties such as this, we’ll need to rearrange the values from clockwise (top, right, bottom, left) to counterclockwise (top, left, bottom, right).